kube-runtime/src/controller/runner.rs

```diff
@@ -32,35 +44,70 @@ where
             scheduler,
             run_msg,
             slots: FutureHashMap::default(),
+            ready_to_execute_after: future::ready(Ok(())).fuse(),
+            is_ready_to_execute: false,
+            stopped: false,
+        }
+    }
+
+    /// Wait for `ready_to_execute_after` to complete before starting to run any scheduled tasks.
+    ///
+    /// `scheduler` will still be polled in the meantime.
+    pub fn delay_tasks_until<Ready, ReadyErr>(
```

Comment:
minor note: you caught me out with `pub` and proper docs, but this is not actually re-exported (which is good). i didn't expect people to actually use `Runner` themselves (otoh, i didn't expect people use `Scheduler` either but that's pub)

Reply:
Mostly because I caught myself here too..